sound/soc/sof/intel/hda-stream.c

```diff
@@ -1082,7 +1082,7 @@ snd_pcm_uframes_t hda_dsp_stream_get_position(struct hdac_stream *hstream,
 		break;
 	}
 
-	if (pos >= hstream->bufsize)
+	if (pos > hstream->bufsize)
```

Comment:
you could make this conditional and apply it to non cyclic (one shot) transfers only, like the fw loading, library loading.